src/api/githubService.js

```diff
@@ -0,0 +1,286 @@
+          message: workflow.head_commit.message,
+          timestamp: workflow.head_commit.timestamp,
+          sha: workflow.sha,
+          event: workflow.event,
+          node_id: workflow.node_id,
+          url: workflow.url,
+          html_url: workflow.html_url,
+          created: workflow.created_at,
+          updated: workflow.updated_at,
+          run_attempt: workflow.run_attempt,
+          referenced_wf: workflow.referenced_workflows[0],
+          triggering_actor: workflow.triggering_actor.login,
+          jobs_url: workflow.jobs_url,
+          logs_url: workflow.logs_url,
+          workflow_url: workflow.workflow_url,
+        };
+
+        return_data.push(workflow_run);
+      }
+
+      i = i + 1;
+    }
+  } catch (err) {
+    console.log("Unable to fetch getSingleWorkflow", err);
+    return;
+  }
+
+  return return_data;
+};
+
+const _getWorkflowCheckRuns = async (checkSuiteId) => {
+  const workflowCheckRunsApi = `https://api.github.com/repos/mozilla-mobile/mozilla-vpn-client/check-suites/${checkSuiteId}/check-runs`;
+  let return_data = [];
+
+  try {
+    const {
+      data: { check_runs },
+    } = await axios.get(workflowCheckRunsApi);
+    for (let check_run of check_runs) {
+      if (check_run.name !== "Build Test Client") {
+        const run = {
+          id: check_run.id,
+          name: check_run.name,
+          status: check_run.status,
+          conclusion: check_run.conclusion,
+          head_sha: check_run.head_sha,
+          started: check_run.started_at,
+          completed: check_run.completed_at,
+          annotations: check_run.output.annotations_count,
+        };
+
+        return_data.push(run);
+      }
+    }
+  } catch (err) {
+    console.log("Unable to fetch _getWorkflowCheckRuns", err);
+    return;
+  }
+
+  return return_data;
+};
+
+const _getWorkflowCheckRunAnnotation = async (checkRunId) => {
+  const workflowCheckRunAnnotationApi = `https://api.github.com/repos/mozilla-mobile/mozilla-vpn-client/check-runs/${checkRunId}/annotations`;
+  let return_data = [];
+
+  try {
+    const { data } = await axios.get(workflowCheckRunAnnotationApi);
+    for (let run_annotation of data) {
+      const annotation = {
+        message: run_annotation.message,
+        annotation_level: run_annotation.annotation_level,
+        start_line: run_annotation.start_line,
+        end_line: run_annotation.end_line,
+      };
+
+      return_data.push(annotation);
+    }
+  } catch (err) {
+    console.log("Unable to fetch _getWorkflowCheckRunAnnotation", err);
+    return;
+  }
+
+  return return_data;
+};
+
+/// full api
+export const getFullTestWorkflowReport = async () => {
+  let return_data = {
+    fetched: moment().format().valueOf(),
+    workflow_runs_data: [],
+  };
+
+  if (await _getLocalData()) {
+    const localData = await _getDataFromLocal();
+    return_data.fetch_message =
+      "Its too early to request again, clear expiretime";
```

Comment:
why this message?

Reply:
thought it'd be useful to someone looking at the page incase they'd like new data before the scheduled times.